packages/lesswrong/components/forumEvents/ForumEventPoll.tsx

```diff
@@ -259,33 +279,35 @@ export const ForumEventPoll = ({event, postId, classes}: {
    * When the user clicks the "x" icon, or when a logged out user tries to vote,
    * delete their vote data
    */
-  const clearVote = useCallback((e?: React.MouseEvent) => {
+  const clearVote = useCallback(async (e?: React.PointerEvent) => {
     e?.stopPropagation()
     setVotePos(defaultVotePos)
     setCurrentUserVote(null)
-    if (currentUser) {
-      void removeVote({variables: {forumEventId: event._id}})
+    if (currentUser && event) {
       setVoteCount(count => count - 1)
+      await removeVote({variables: {forumEventId: event._id}})
+      refetch?.()
```

Comment:
I added refetches after the user's vote changes so that when they navigate between home and post pages, their vote data is properly updated